fedbiomed/transport/node_agent.py

```diff
@@ -158,10 +173,13 @@ async def send_async(
                             "as DISCONNECTED soon if no request received.")
 
         # Updates replies
+        #
+        # Note: as forwarded messages don't have a `request_id` field we don't have to test
+        # if this is an OverlayMessage but check whether the field exists
         async with self._replies_lock:
             # update replies only for (1) request-response messages
             # (2) that are not yet registered as pending request
-            if message.request_id and message.request_id not in self._replies:
+            if hasattr(message, 'request_id') and message.request_id and message.request_id not in self._replies:
```

Comment:
Sorry @srcansiz I don't understand the "does not reach to this point" (am I missing smthg ?).

`_GrpcAsyncServer._on_forward()` => `_GrpcAsyncServer.send()` => `NodeAgentAsync.send_async()`
would reach this point ?

Another reason could be: it makes the code future proof if we introduce `ResearcherMessages` that are not request reply :-)

Reply:
Because if message is an overlay message the method executes `_on_forward` and `return` see below. I thought that overlay message has no `request_id` and that's why the method checks if `request_id` existing. 


```python
 if isinstance(message, OverlayMessage):
            if asyncio.iscoroutinefunction(self._on_forward):
                await self._on_forward(message)
            else:
                logger.warning(f"No function defined for handling overlay message received from {self._id}. "
                               "Discard message.")
            return


```

> Another reason could be: it makes the code future proof if we introduce ResearcherMessages that are not request reply :-)

Currently we do not have a message that does not have `request_id`. I suggest to implement it once we have such use case.

Reply:
@srcansiz sorry I still don't get it ... what am I missing ?
*During* the `on_forward`, thus before the `return`, the `NodeAgentAsync.send_async()` is called - so it looks like the point is reached.